helix-term/src/commands.rs

```diff
@@ -3896,6 +3901,105 @@ fn surround_delete(cx: &mut Context) {
     })
 }
 
+#[derive(Eq, PartialEq)]
+enum ShellBehavior {
+    Replace,
+    Insert,
+    Append,
+    Filter,
+    None,
+}
+
+fn shell_pipe(cx: &mut Context) {
+    shell(cx, "pipe:", true, ShellBehavior::Replace);
+}
+
+fn shell_pipe_ignore(cx: &mut Context) {
+    shell(cx, "pipe:", true, ShellBehavior::None);
+}
+
+fn shell_insert(cx: &mut Context) {
+    shell(cx, "command:", false, ShellBehavior::Insert);
+}
+
+fn shell_append(cx: &mut Context) {
+    shell(cx, "command:", false, ShellBehavior::Append);
+}
+
+fn shell_filter(cx: &mut Context) {
+    shell(cx, "filter:", true, ShellBehavior::Filter);
```

Comment:
I'm unsure about these prompts. Should they specify that they expect shell commands? I especially `command:` should be changed.

Reply:
Kakoune seems to use 
- `pipe`/`pipe-to` for `|` `alt-|`
- `keep-pipe` for `$`
- `insert-output`/`append-output` for `!` `alt-!`

Reply:
> `pipe-to`
I thought `A-|` just ignored the output of the commands?

Reply:
yeah, kak calls that "pipe to"